flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManager.java

```diff
@@ -133,7 +139,7 @@
 	private final FatalErrorHandler fatalErrorHandler;
 
 	/** The slot manager maintains the available slots. */
-	private final SlotManager slotManager;
+	protected final SlotManager slotManager;
```

Comment:
I think it may be better to have a protected `setAllowedSlotResourceProfiles` method in the ResourceManager than making `slotManager` to be protected directly.